### PR TITLE
Small changes

### PR DESCRIPTION
**Pull request by mosuem**
While I also like composition over inheritance, I thought this would be a good use of polymorphism. This is absolutely optional, just an idea.

### CODE REVIEW

File: packages/corpus/lib/report.dart [hunk 1]
```diff
@@ -9,37 +9,55 @@ import 'package:corpus/pub.dart';
...
 
-  String get type => isDartLibrary ? 'dart' : 'package';
+  @override
+  String get type => 'package';
+
+  @override
+  Stream<PackageInfo> getPackages(Pub pub) => pub.popularDependenciesOf(name);
+}
+
+class DartLibrary extends ReportTarget {
```
Comment (devoncarew) [inline packages/corpus/lib/report.dart:53]: `DartLibrary` ==> `DartLibraryTarget`?

File: packages/corpus/lib/report.dart [hunk 2]
```diff
@@ -58,11 +76,11 @@ class Report {
     buf.writeln();
     buf.writeln('## General info');
     buf.writeln();
-    if (reportTarget.isDartLibrary) {
+    if (reportTarget is DartLibrary) {
       buf.writeln('https://api.dart.dev/dart-${reportTarget.name}/'
           'dart-${reportTarget.name}-library.html');
     } else {
-      buf.writeln(reportTarget.description);
+      buf.writeln((reportTarget as PackageTarget).description);
```
Comment (devoncarew) [inline packages/corpus/lib/report.dart:83]: If `description` were a nullable property of ReportTarget we could avoid this cast.

**Review by devoncarew — APPROVED** [review]
Looks good! This is an improvement over the multiple if statements I had. One suggestion about a class rename.

File: packages/corpus/lib/report.dart (hunk 2 above)
Comment (mosuem) [reply packages/corpus/lib/report.dart:83]: Added the type check to clarify, but I would rather not make this a property of `ReportTarget` as a `DartLibraryTarget` really has no description (so far).

**Pull request merged**